projects/plugins/jetpack/modules/subscriptions/subscribe-modal/class-jetpack-subscribe-modal.php

```diff
@@ -0,0 +1,236 @@
+					'class' => array(),
+				),
+				'input'  => array(
+					'id'          => array(),
+					'type'        => array(),
+					'name'        => array(),
+					'value'       => array(),
+					'class'       => array(),
+					'style'       => array(),
+					'placeholder' => array(),
+					'required'    => array(),
+				),
+				'button' => array(
+					'type'  => array(),
+					'class' => array(),
+					'style' => array(),
+				),
+			);
+
+			?>
+				<div class="jetpack-subscribe-modal">
+					<div class="jetpack-subscribe-modal__modal-content">
+						<?php
+						foreach ( $blocks as $block ) {
+							echo wp_kses( render_block( $block ), $allowed_html );
+						}
+						?>
+						<Dashicon icon="minus" className="jp-idc__idc-screen__card-action-separator" />
+					</div>
+				</div>
+			<?php
+		}
+	}
+
+	/**
+	 * Enqueues JS to load modal.
+	 *
+	 * @return void
+	 */
+	public function wpcom_enqueue_subscribe_modal_assets() {
+		if ( $this->should_enable_subscriber_modal() && ! is_admin() ) {
+			wp_enqueue_style( 'subscribe-modal-css', plugins_url( 'subscribe-modal.css', __FILE__ ), array(), JETPACK__VERSION );
+			wp_enqueue_script( 'subscribe-modal-js', plugins_url( 'subscribe-modal.js', __FILE__ ), array(), JETPACK__VERSION, true );
+		}
+	}
+
+	/**
+	 * Returns true if we should enable Newsletter content.
+	 * This is currently limited to lettre theme or newsletter sites.
+	 * We could open it to all themes or site intents.
+	 *
+	 * @return bool
+	 */
+	public function should_enable_subscriber_modal() {
+		$is_lettre          = get_option( 'stylesheet' ) === 'lettre';
+		$is_newsletter_site = get_option( 'site_intent' ) === 'newsletter';
+		$is_modal_enabled   = get_option( 'sm_enabled' ) || false;
+		$is_fse_theme       = wp_is_block_theme();
+
+		return ( $is_lettre || $is_newsletter_site ) && $is_fse_theme && $is_modal_enabled;
+	}
+
+	/**
+	 * Returns the slug for the Subcribe template.
+	 *
+	 * @return string
+	 */
+	public function get_subscribe_template_slug() {
+		return 'subscribe-modal';
+	}
+
+	/**
+	 * Returns the initial content of the Subscribe Modal template.
+	 * This can then be edited by the user.
+	 *
+	 * @return string
+	 */
+	public function get_subscribe_template_content() {
+		return '<!-- wp:group {"style":{"spacing":{"padding":{"top":"50px","bottom":"50px","left":"20px","right":"20px"}}},"layout":{"type":"constrained"}} -->
+		<div class="wp-block-group" style="padding-top:50px;padding-right:20px;padding-bottom:50px;padding-left:20px"><!-- wp:group {"style":{"dimensions":{"minHeight":"0px"},"spacing":{"blockGap":"8px"}},"layout":{"type":"flex","flexWrap":"wrap","justifyContent":"center"}} -->
+		<div class="wp-block-group" style="min-height:0px"><!-- wp:heading {"style":{"typography":{"fontStyle":"normal","fontWeight":"600","fontSize":"26px"},"layout":{"selfStretch":"fit","flexSize":null}}} -->
+		<h2 class="wp-block-heading" style="font-size:26px;font-style:normal;font-weight:600">Discover more from</h2>
```

Comment:
should we make these translatable?

Reply:
Since we weren't sure, I just tested this. It works. Having the translation methods as part of the markup here doesn't negatively affect creating the template. So I've pushed the change adding translations.